src/cipher.rs

```diff
@@ -509,6 +511,97 @@ impl rustls_root_cert_store {
     }
 }
 
+/// A builder for a `rustls_client_cert_verifier`. This builder object can be used to configure
+/// certificate revocation lists, and then turned into a `rustls_client_cert_verifier` once ready.
+pub struct rustls_client_cert_verifier_builder {
+    _private: [u8; 0],
+}
+
+// XXX: contained value is consumed even on error, so this can contain None. but the caller
+// still needs to free it
```

Comment:
This is Good, Actually.

There's a problem where Rust's "into" patterns can lead to really complicated lifecycle management on the C side. I wrote up the problem here: https://github.com/rustls/rustls-ffi/issues/244, and implemented the solution here: https://github.com/rustls/rustls-ffi/pull/243/files#diff-5451b831ef642c0ec3aab8375dd75414f81b677f5f638a3d44d1ff1d226117daR59-R61. The basic idea is that if you created something, you have to free it, whether you (a) converted it using an "into' or builder call, (b) failed to convert it using an "into" or builder call, or (c) errored out before trying to convert it.

In the Acceptor PR I did `RustType = Option<X>`; here you have a newtype. I think the `Option<X>` is slightly nicer but don't feel super stongly; however, we should do the same in both places.

Reply:
I switched to using `Option<X>` instead of a newtype. I agree it seems slightly nicer than the newtype :+1: